charter.md

```diff
@@ -26,4 +26,19 @@ Members of the style team are nominated by existing members. All existing member
+
+1. Representation and Affiliation: The style team values representation from various parts of the Rust organization and ecosystem. We aim to include members affiliated with different areas of the rust ecosystem as best as possible, to ensure a well-rounded perspective.
+2. Familiarity with Rust: Prospective members should have a general familiarity with Rust, including a solid understanding of its constructs, how they interact, and the overall language ecosystem. This knowledge is crucial for making informed decisions regarding style and formatting guidelines.
+3. Interest in Rust Style: Members should have a genuine interest in Rust style and a willingness to contribute to the team's efforts. It is important that members are not solely focused on advocating for their personal style preferences but are dedicated to promoting a collective and consensus-based approach.
+4. Effective Collaboration Skills: Successful candidates will have a proven track record of effective collaboration, showcasing their ability to work together with others, consider diverse perspectives, navigate conflicts, and address emotional challenges in a constructive manner.
+5. Familiarity with Rust Project: Familiarity with the Rust project's structure, culture, and expectations upon maintainers is desirable. This knowledge helps members navigate the ecosystem more effectively and align their style decisions with the project's broader goals.
+6. Consent to Principles and Aims: Members are expected to align with and consent to the principles and aims of the style team, as outlined in the [Rust style guide principles](https://github.com/rust-lang/rust/blob/master/src/doc/style-guide/src/principles.md) and the style team's [charter](https://github.com/rust-lang/style-team/blob/style-policy/charter.md).
+7. Pragmatic Approach: Members should prioritize pragmatism over dogmatism, even when dealing with style and formatting aspects they feel particularly strongly about. The ability to make practical and reasonable decisions that benefit the community at large is essential.
+8. Openness to Change and Learning: Members should demonstrate a willingness to seek out and receive new information, be open to changing their opinions based on evidence and rationale, and adapt their perspectives accordingly.
+9. Effective Communication in Charged Conversations: The ability to navigate charged conversations and respond constructively and empathetically to evocative commentary is important. Members should be able to handle passionate discussions about code style, recognize miscommunication and misunderstandings, and promote empathy and listening when discussing matters within the purview of the style team.
+10. Interest in Language and Syntax: Members should show interest in tracking changes to the Rust language and syntax. Keeping up-to-date with the latest developments helps inform style decisions
+
```

Comment:
```suggestion
11. Ability to make non-universally-satisfying decisions: Members must be able to participate in consensus choices that, by their nature, will never be able to simultaneously satisfy the entire Rust community. The style team values having a default Rust style, even though the default Rust style will never be perfect for everyone. Members must be able to acknowledge this and still help the team make the best choices it can.

```

I was inspired to add this because many parts of the Rust project tend to treat tradeoffs as challenges: why *can't* we satisfy all of these values at once, are they *really* mutually exclusive, we can do better. Style tries to do the same when we can, but style is particularly likely to have to make tradeoffs that can't possibly satisfy everyone. It's important that people know that going in, and not burn out as a result of it.

Reply:
I like the idea here but am experiencing some hesitation and trying to introspect on why. I think it's ultimately a preference of how we should frame this, so it's not an objection. I will share my alternative framing and thoughts and see what y'all think before making any final decisions for this item.

This is essentially the same point as the "overlap of tolerance vs. overlap of preference" idea that I often return to. Usually, there are many acceptable solutions, but there is rarely a single solution that everyone prefers[^1]. I think the part that I'm reacting negatively to is the wording "non-universally-satisfying decisions." I worry that this is ambiguous and that people could misinterpret it as meaning "decisions that someone objected to," but then again, when I ask myself what the difference is between "not satisfying" and "preference," it doesn't feel like a big one, so I feel like it's saying the same thing, just not the way I would have said it which makes me anxious I guess(??).

When I tried considering alternatives, I concluded that this is ultimately describing the consent decision-making process and focusing on objections over preference.

What do you think of this alternative?

```suggestion
11. Familiarity with consent decision-making process: Members must be able to participate in consensus choices that, by their nature, will never be able to simultaneously satisfy the entire Rust community. The style team values having a default Rust style, even though the default Rust style will never be perfect for everyone. Members must be able to acknowledge this and still help the team make the best choices it can.
```

[^1]: I think perfectionism also ties into this a bit, in that people with perfectionist tendencies are going to be much more likely to want to aim for solutions that satisfy everyone.

Reply:
I feel like this is not inherently a property of consent decision-making. One could have a team that operates by consent, but regularly finds itself unable to make a decision because people on the team object to each possibility raised, and do not consent to the outcomes.

Imagine trying to decide tabs vs spaces, and assume the members of the team disagree. They need to still be able to agree that picking *one* of those is better than leaving it unspecified. But that takes recognizing that they are implicitly deciding to leave it unspecified by falling to consent to either way of specifying it.

I don't object to tweaking the wording to make the intent clear. And I do believe that good decision-making requires understanding the impact of the null action and acknowledging that the team failing to consent to anything else will result in the null action; this is something many parts of the project would also benefit from. But I don't think this particular retitling makes that clear, and I do think it's important to emphasize for the style team *explicitly*.

What do you think about a title like "Acknowledging that failing to make a choice is also a choice", or similar?

Reply:
I agree that "consensus" is different than "consent".

Reply:
> I also want to note that that "consensus" is different than "consent".

clarifying question: is this referring to the "consensus choices" bit of the suggestion from Josh? And does this mean you'd like us to change the wording (e.g. "consent choices" or maybe just "decisions")?

Reply:
I'm mostly noting that Jane's revised title means something different than Josh's title (at least to me), and tying it back to Josh's first sentence:

> I feel like this is not inherently a property of consent decision-making. One could have a team that operates by consent, but regularly finds itself unable to make a decision because people on the team object to each possibility raised, and do not consent to the outcomes.

Reply:
I am responding out of order because my response to the first section is more of an aside and not the main point I want to focus on:[^1]

> Imagine trying to decide tabs vs spaces, and assume the members of the team disagree. They need to still be able to agree that picking _one_ of those is better than leaving it unspecified. But that takes recognizing that they are implicitly deciding to leave it unspecified by falling to consent to either way of specifying it.
> 
> I don't object to tweaking the wording to make the intent clear. And I do believe that good decision-making requires understanding the impact of the null action and acknowledging that the team failing to consent to anything else will result in the null action; this is something many parts of the project would also benefit from. But I don't think this particular retitling makes that clear, and I do think it's important to emphasize for the style team _explicitly_.
> 
> What do you think about a title like "Understanding that failing to make a choice is also a choice", or similar?

I feel like this is getting at a different issue than what I originally got from the suggestion. Originally, I thought we were talking about how to make decisions effectively/efficiently. How do we make sure we don't bikeshed unnecessarily and we make forward progress that meets people's needs but doesn't waste time trying to meet everyone's preference? The above quote seems to be focused more on the risks of deadlock and how to handle it. I don't disagree with the expressed belief[^2], but I want to ensure we're not confusing them and that we only put them in the same bullet point if it makes sense.

On the connection between those two issues: I feel like these are related in so far as they're both skills related to decision making, making them quickly and effectively and handling difficult edge cases with decision making.

> I feel like this is not inherently a property of consent decision-making. One could have a team that operates by consent, but regularly finds itself unable to make a decision because people on the team object to each possibility raised, and do not consent to the outcomes.

I find myself doubtful that this is a problem in practice with the consent decision-making framing. I feel like this failure mode is much more common in consensus decision-making, where people object or do not approve based on preference. It's undoubtedly still theoretically possible for a team to deadlock in a consent decision-making framing, but I think focusing on that possibility rather than how much consent decision-making does to address that specific concern already is missing the forest for the trees.

---

So, tying it all together. The goal here is to establish expectations for members so we can recruit new people and help onboarding them with clarity around their new role. For each of these points, I'm asking myself if listing them in the membership criteria would help with that goal and help people be effective members of the style team. I feel like the former and meta points about effective decision-making and decision-making skills are very relevant; we should highlight these skills. 

Does "highlighting the potential costs of deadlock" contribute to this goal? I do not think it does; I believe it would run counter to that goal by priming people to focus on the negatives and failure modes around decision-making, which could contribute to a culture of fear and distrust[^3]. I think this connects back to an idea we discussed a while ago about setting up a shared governance library where we can centrally define things like "rounds" and "consent decision-making" in the context of the Rust project. This point would be appropriate for such a larger document about decision-making, but I feel that it's too nuanced to be helpful in this context.

My counter-proposal is to take a step back from the specific points and focus on the overall point that encapsulates everything we've discussed. Half-assed strawman proposal "Decision-making skills: members should be familiar with the consensus and consent decision-making framings, their trade-offs, and their potential failure modes." with the expectation being that we will add links to supplemental documents as they become available which detail those trade-offs and failure modes, including the bit about failures to make a decision.

[^1]: I'm feeling very meta here, trying to introspect on my beliefs because I can feel how emotional my motivations are in this one, and I want to make sure I'm taking the time to understand where these emotions are coming from and that they're well founded. Apologies for the length, and thank you for understanding ^_^''
[^2]: "that it's important for teams to recognize when failing to make a decision is itself a decision" or put another way, "awareness of the status quo bias in our decision-making methods and its potential costs."
[^3]: focusing primarily on how decision-making breaks down rather than focusing on how to prevent decision-making breakdown, building an expectation of dysfunction and disagreement, contributing to skepticism of our decision-making processes, leading people to be less engaged and interested in learning about those processes and building up their decision-making skills.